diff --git a/build/BUILD.gn b/build/BUILD.gn
index a31d5b4..3653af1 100644
--- a/build/BUILD.gn
+++ b/build/BUILD.gn
@@ -14,8 +14,8 @@
 import("//build/config/config.gni")
 import("//build/ohos.gni")
 import("//foundation/arkui/ace_engine/ace_config.gni")
-import(
-    "//foundation/arkui/ace_engine/frameworks/core/components_ng/components_plugin.gni")
+# import(
+#     "//foundation/arkui/ace_engine/frameworks/core/components_ng/components_plugin.gni")
 import("//foundation/arkui/ace_engine/interfaces/napi/kits/napi_lib.gni")
 import("//plugins/plugin_lib.gni")
 import("config.gni")
@@ -157,38 +157,38 @@ foreach(module, common_plugin_libs) {
 }
 
 # components target
-component_plugin_modules = []
-foreach(module, component_plugins) {
-  sub_name = string_replace(module, "_", "")
-  module_name = "arkui_$sub_name"
-  lib_name = "lib" + module_name
-  ohos_shared_library(module_name) {
-    ldflags = [
-      "-Wl",
-      "-install_name",
-      "@rpath/${lib_name}.framework/$lib_name",
-    ]
-    deps = [ "$ace_root/frameworks/core/components_ng/pattern/${module}:components_plugins_${module}_source_ios" ]
-    deps += [ ":arkui_ios" ]
-    frameworks = [
-      "AVFoundation.framework",
-      "Foundation.framework",
-      "UIKit.framework",
-      "WebKit.framework",
-    ]
-
-    # relative_install_dir = "module"
-    subsystem_name = "arkui"
-    part_name = "arkui_components"
-  }
-  ohos_combine_darwin_framework(lib_name) {
-    deps = [ ":$module_name" ]
-
-    subsystem_name = "arkui"
-    part_name = "arkui_components"
-  }
-  component_plugin_modules += [ ":${lib_name}" ]
-}
+# component_plugin_modules = []
+# foreach(module, component_plugins) {
+#   sub_name = string_replace(module, "_", "")
+#   module_name = "arkui_$sub_name"
+#   lib_name = "lib" + module_name
+#   ohos_shared_library(module_name) {
+#     ldflags = [
+#       "-Wl",
+#       "-install_name",
+#       "@rpath/${lib_name}.framework/$lib_name",
+#     ]
+#     deps = [ "$ace_root/frameworks/core/components_ng/pattern/${module}:components_plugins_${module}_source_ios" ]
+#     deps += [ ":arkui_ios" ]
+#     frameworks = [
+#       "AVFoundation.framework",
+#       "Foundation.framework",
+#       "UIKit.framework",
+#       "WebKit.framework",
+#     ]
+
+#     # relative_install_dir = "module"
+#     subsystem_name = "arkui"
+#     part_name = "arkui_components"
+#   }
+#   ohos_combine_darwin_framework(lib_name) {
+#     deps = [ ":$module_name" ]
+
+#     subsystem_name = "arkui"
+#     part_name = "arkui_components"
+#   }
+#   component_plugin_modules += [ ":${lib_name}" ]
+# }
 
 # Install packages
 group("ace_packages") {
diff --git a/build/config.gni b/build/config.gni
index 7b93e88..388ae8a 100644
--- a/build/config.gni
+++ b/build/config.gni
@@ -42,7 +42,7 @@ enable_rosen_backend = true
 use_external_icu = "static"
 pixel_map_support = true
 video_components_support = true
-use_components_lib = true
+use_components_lib = false
 web_components_support = true
 video_texture_support = true
 
diff --git a/build/libarkui_ios.txt b/build/libarkui_ios.txt
index 21b8999..29db11c 100644
--- a/build/libarkui_ios.txt
+++ b/build/libarkui_ios.txt
@@ -126,18 +126,6 @@ __ZN4OHOS3Ace14AnimationUtils11AddKeyFrameEfRKNS0_6RefPtrINS0_5CurveEEERKNSt3__1
 __ZN4OHOS3Ace14AnimationUtils21OpenImplicitAnimationERKNS0_15AnimationOptionERKNS0_6RefPtrINS0_5CurveEEERKNSt3__18functionIFvvEEE
 __ZN4OHOS3Ace14AnimationUtils22CloseImplicitAnimationEv
 __ZN4OHOS3Ace14AnimationUtils7AnimateERKNS0_15AnimationOptionERKNSt3__18functionIFvvEEESA_SA_
-__ZN4OHOS3Ace14NapiAsyncEvnet4CallEiPP12napi_value__
-__ZN4OHOS3Ace14NapiAsyncEvnet6GetEnvEv
-__ZN4OHOS3Ace14NapiAsyncEvnetC1EP10napi_env__P12napi_value__
-__ZN4OHOS3Ace14NapiAsyncEvnetD1Ev
-__ZN4OHOS3Ace15CommonNapiUtils10CreateNullEP10napi_env__
-__ZN4OHOS3Ace15CommonNapiUtils10ParseColorEP10napi_env__P12napi_value__RNS0_5ColorE
-__ZN4OHOS3Ace15CommonNapiUtils11CreateArrayEP10napi_env__
-__ZN4OHOS3Ace15CommonNapiUtils11CreateInt32EP10napi_env__i
-__ZN4OHOS3Ace15CommonNapiUtils12GetValueTypeEP10napi_env__P12napi_value__
-__ZN4OHOS3Ace15CommonNapiUtils18GetDimensionResultEP10napi_env__P12napi_value__RNS0_13CalcDimensionE
-__ZN4OHOS3Ace15CommonNapiUtils18SetSelementToArrayEP10napi_env__P12napi_value__iS5_
-__ZN4OHOS3Ace15CommonNapiUtils7GetBoolEP10napi_env__P12napi_value__
 __ZN4OHOS3Ace2NG13InputEventHubC1ERKNS0_7WeakPtrINS1_8EventHubEEE
 __ZN4OHOS3Ace2NG14LayoutProperty15UpdateAlignmentENS0_9AlignmentE
 __ZN4OHOS3Ace2NG14LayoutProperty20UpdateLayoutPropertyEPKS2_
@@ -167,7 +155,6 @@ __ZN4OHOS3Ace8JsonUtil6CreateEb
 __ZN4OHOS3Ace9Alignment6CENTERE
 __ZN4OHOS3Ace9ToRSColorERKNS0_11LinearColorE
 __ZN4OHOS3Ace9ToRSColorERKNS0_5ColorE
-__ZN4OHOS5Rosen7Drawing10MaskFilter20CreateBlurMaskFilterENS1_8BlurTypeEf
 __ZN4OHOS5Rosen7Drawing12ShaderEffect20CreateLinearGradientERKNS1_6PointFES5_RKNSt3__16vectorIjNS6_9allocatorIjEEEERKNS7_IfNS8_IfEEEENS1_8TileModeE
 __ZN4OHOS5Rosen7Drawing3Pen11SetCapStyleENS2_8CapStyleE
 __ZN4OHOS5Rosen7Drawing3Pen12SetAntiAliasEb
@@ -211,9 +198,6 @@ __ZTVN4OHOS3Ace2NG18TouchEventActuatorE
 __Z28napi_module_with_js_registerP19napi_module_with_js
 __ZN4OHOS3Ace10LogWrapper5GetIdEv
 __ZN4OHOS3Ace12PipelineBase17GetCurrentContextEv
-__ZN4OHOS3Ace15CommonNapiUtils10CreateNullEP10napi_env__
-__ZN4OHOS3Ace15CommonNapiUtils10ParseColorEP10napi_env__P12napi_value__RNS0_5ColorE
-__ZN4OHOS3Ace15CommonNapiUtils22GetStringFromValueUtf8EP10napi_env__P12napi_value__
 __ZN4OHOS3Ace2NG14LayoutProperty15UpdateAlignmentENS0_9AlignmentE
 __ZN4OHOS3Ace2NG15CreateIdealSizeERKNS1_17LayoutConstraintTIfEENS0_4AxisENS1_11MeasureTypeEb
 __ZN4OHOS3Ace2NG15ModifierAdapter14RemoveModifierEi
@@ -231,6 +215,19 @@ __ZN4OHOS3Ace5Color5BLACKE
 __ZN4OHOS3Ace5Color5WHITEE
 __ZN4OHOS3Ace8JsonUtil6CreateEb
 __ZN4OHOS3Ace9Alignment6CENTERE
+__ZN4OHOS3Ace12SpringMotionC1EdddRKNS0_6RefPtrINS0_14SpringPropertyEEE
+__ZN4OHOS3Ace8Animator10PlayMotionERKNS0_6RefPtrINS0_6MotionEEE
+__ZNK4OHOS3Ace14ThemeConstants15GetPluralStringEji
+__ZNK4OHOS3Ace14ThemeConstants15GetStringByNameERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEE
+__ZNK4OHOS3Ace14ThemeConstants18GetDimensionByNameERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEE
+__ZNK4OHOS3Ace14ThemeConstants21GetPluralStringByNameERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEi
+__ZN4OHOS3Ace5Color16ParseColorStringENSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEERS1_j
+__ZNK4OHOS3Ace14ThemeConstants6GetIntEj
+__ZNK4OHOS3Ace14ThemeConstants8GetColorEj
+__Z35napi_set_promise_rejection_callbackP10napi_env__P10napi_ref__S2_
+__ZN4OHOS3Ace12EngineHelper18GetDefaultDelegateEv
+__ZN12NativeEngine19HasListeningCounterEv
+__Z25napi_create_limit_runtimeP10napi_env__PS0_
 __ZN4OHOS5Rosen7Drawing6Bitmap5BuildEiiRKNS1_12BitmapFormatEi
 __ZN4OHOS5Rosen7Drawing6BitmapC1Ev
 __ZN4OHOS5Rosen7Drawing6BitmapD1Ev
diff --git a/entrance/BUILD.gn b/entrance/BUILD.gn
index bd2d217..77379f8 100644
--- a/entrance/BUILD.gn
+++ b/entrance/BUILD.gn
@@ -58,6 +58,7 @@ template("ace_ios_standard_source_set") {
       "display_info.mm",
       "download_manager.cpp",
       "virtual_rs_window.mm",
+      "foldable_window_ios.cpp",
     ]
 
     if (target_cpu == "arm64") {
diff --git a/entrance/virtual_rs_window.h b/entrance/virtual_rs_window.h
index 42c59fd..1ac53d9 100644
--- a/entrance/virtual_rs_window.h
+++ b/entrance/virtual_rs_window.h
@@ -137,6 +137,7 @@ public:
     bool CreateVSyncReceiver(std::shared_ptr<AppExecFwk::EventHandler> handler);
     void RequestNextVsync(std::function<void(int64_t, void*)> callback);
 
+    virtual void FlushFrameRate(int32_t rate) {}
     virtual void RequestVsync(const std::shared_ptr<VsyncCallback>& vsyncCallback);
 
     void CreateSurfaceNode(void* layer);
diff --git a/osal/BUILD.gn b/osal/BUILD.gn
index 1df2110..f568d99 100644
--- a/osal/BUILD.gn
+++ b/osal/BUILD.gn
@@ -27,6 +27,7 @@ template("ace_osal_ios_source_set") {
       "ace_engine_ext.cpp",
       "app_bar_helper_impl.cpp",
       "download_manager_ios.cpp",
+      "download_manager_ios_v2.cpp",
       "drag_window.cpp",
       "event_report.cpp",
       "exception_handler.cpp",
@@ -52,6 +53,10 @@ template("ace_osal_ios_source_set") {
       "socperf_client_impl.cpp",
       "subwindow.cpp",
       "system_properties.cpp",
+      "advance/data_detector_mgr.cpp",
+      "advance/image_analyzer_mgr.cpp",
+      "time_event_proxy_ios.cpp",
+      "view_data_wrap_impl.cpp",
     ]
 
     cflags_cc = [
diff --git a/osal/event_report.cpp b/osal/event_report.cpp
index c5bbfe1..a2b4626 100644
--- a/osal/event_report.cpp
+++ b/osal/event_report.cpp
@@ -62,4 +62,8 @@ void EventReport::ReportEventComplete(DataBase& data) {}
 void EventReport::ReportEventJankFrame(DataBase& data) {}
 
 void EventReport::ReportJankFrameApp(JankInfo& info) {}
+
+void EventReport::ReportDoubleClickTitle(int32_t stateChange) {}
+
+void EventReport::ReportClickTitleMaximizeMenu(int32_t maxMenuItem, int32_t stateChange) {}
 } // namespace OHOS::Ace
diff --git a/osal/ios_trace.cpp b/osal/ios_trace.cpp
index 51fba4b..3ddb123 100644
--- a/osal/ios_trace.cpp
+++ b/osal/ios_trace.cpp
@@ -58,9 +58,9 @@ void AceTraceEnd()
     printf(">>>>> trace end! %s\n", GetTraceTimeStamp().c_str());
 }
 
-void AceAsyncTraceBegin(int32_t taskId, const char* name) {}
+void AceAsyncTraceBegin(int32_t taskId, const char* name, bool isAnimationTrace) {}
 
-void AceAsyncTraceEnd(int32_t taskId, const char* name) {}
+void AceAsyncTraceEnd(int32_t taskId, const char* name, bool isAnimationTrace) {}
 
 void AceCountTrace(const char *key, int32_t count)
 {
diff --git a/osal/resource_adapter_impl.cpp b/osal/resource_adapter_impl.cpp
index 362e42b..83a1f13 100644
--- a/osal/resource_adapter_impl.cpp
+++ b/osal/resource_adapter_impl.cpp
@@ -177,7 +177,7 @@ RefPtr<ResourceAdapter> ResourceAdapter::CreateNewResourceAdapter(
     return nullptr;
 }
 
-void ResourceAdapterImpl::UpdateConfig(const ResourceConfiguration& config)
+void ResourceAdapterImpl::UpdateConfig(const ResourceConfiguration& config, bool themeFlag)
 {
     LOGI("UpdateConfig ori=%{public}d, dpi=%{public}d, device=%{public}d, colorMode=%{public}d,",
         config.GetOrientation(), config.GetDensity(), config.GetDeviceType(), config.GetColorMode());
@@ -188,7 +188,7 @@ void ResourceAdapterImpl::UpdateConfig(const ResourceConfiguration& config)
          "colorMode=%{public}d, inputDevice=%{public}d",
         resConfig->GetDirection(), resConfig->GetScreenDensity(), resConfig->GetDeviceType(), resConfig->GetColorMode(),
         resConfig->GetInputDevice());
-    resourceManager_->UpdateResConfig(*resConfig);
+    resourceManager_->UpdateResConfig(*resConfig, themeFlag);
 }
 
 RefPtr<ThemeStyle> ResourceAdapterImpl::GetTheme(int32_t themeId)
diff --git a/osal/resource_adapter_impl.h b/osal/resource_adapter_impl.h
index 74a3122..16f2fe9 100644
--- a/osal/resource_adapter_impl.h
+++ b/osal/resource_adapter_impl.h
@@ -33,7 +33,7 @@ public:
     ~ResourceAdapterImpl() override = default;
 
     void Init(const ResourceInfo& resourceInfo) override;
-    void UpdateConfig(const ResourceConfiguration& config) override;
+    void UpdateConfig(const ResourceConfiguration& config, bool themeFlag = false) override;
 
     RefPtr<ThemeStyle> GetTheme(int32_t themeId) override;
 
diff --git a/osal/system_properties.cpp b/osal/system_properties.cpp
index 885c711..d7c86b3 100644
--- a/osal/system_properties.cpp
+++ b/osal/system_properties.cpp
@@ -33,6 +33,8 @@ bool SystemProperties::isRound_ = false;
 bool SystemProperties::isDeviceAccess_ = false;
 int32_t SystemProperties::deviceWidth_ = 0;
 int32_t SystemProperties::deviceHeight_ = 0;
+int32_t SystemProperties::devicePhysicalWidth_ = 0;
+int32_t SystemProperties::devicePhysicalHeight_ = 0;
 double SystemProperties::resolution_ = 1.0;
 DeviceType SystemProperties::deviceType_ { DeviceType::PHONE };
 DeviceOrientation SystemProperties::orientation_ { DeviceOrientation::PORTRAIT };
@@ -59,7 +61,7 @@ bool SystemProperties::debugEnabled_ = false;
 bool SystemProperties::debugBoundaryEnabled_ = false;
 bool SystemProperties::extSurfaceEnabled_ = true;
 uint32_t SystemProperties::dumpFrameCount_ = 0;
-bool SystemProperties::flutterDecouplingEnabled_ = true;
+bool SystemProperties::layoutTraceEnable_ = false;
 
 void SystemProperties::InitDeviceType(DeviceType type)
 {
@@ -182,4 +184,23 @@ bool SystemProperties::GetResourceDecoupling()
     return false;
 }
 
+int32_t SystemProperties::GetJankFrameThreshold()
+{
+    return 0;
+}
+
+bool SystemProperties::GetTitleStyleEnabled()
+{
+    return false;
+}
+
+std::string SystemProperties::GetCustomTitleFilePath()
+{
+    return UNDEFINED_PARAM;
+}
+
+bool SystemProperties::Is24HourClock()
+{
+    return false;
+}
 } // namespace OHOS::Ace
diff --git a/stage/uicontent/ace_container_sg.cpp b/stage/uicontent/ace_container_sg.cpp
index 17dc7b9..9243126 100644
--- a/stage/uicontent/ace_container_sg.cpp
+++ b/stage/uicontent/ace_container_sg.cpp
@@ -221,7 +221,8 @@ void AceContainerSG::InitializeCallback()
     ACE_DCHECK(aceView_ && taskExecutor_ && pipelineContext_);
     auto weak = AceType::WeakClaim(AceType::RawPtr(pipelineContext_));
     auto instanceId = aceView_->GetInstanceId();
-    auto&& touchEventCallback = [weak, instanceId](const TouchEvent& event, const std::function<void()>& markProcess) {
+    auto&& touchEventCallback = [weak, instanceId](const TouchEvent& event, const std::function<void()>& markProcess,
+                                    const RefPtr<OHOS::Ace::NG::FrameNode>& node) {
         auto context = weak.Upgrade();
         CHECK_NULL_VOID(context);
 
@@ -255,7 +256,8 @@ void AceContainerSG::InitializeCallback()
     };
     aceView_->RegisterKeyEventCallback(keyEventCallback);
 
-    auto&& mouseEventCallback = [weak, instanceId](const MouseEvent& event, const std::function<void()>& markProcess) {
+    auto&& mouseEventCallback = [weak, instanceId](const MouseEvent& event, const std::function<void()>& markProcess,
+                                    const RefPtr<OHOS::Ace::NG::FrameNode>& node) {
         auto context = weak.Upgrade();
         CHECK_NULL_VOID(context);
 
diff --git a/stage/uicontent/ace_view_sg.cpp b/stage/uicontent/ace_view_sg.cpp
index 1ab2a3f..10fd130 100644
--- a/stage/uicontent/ace_view_sg.cpp
+++ b/stage/uicontent/ace_view_sg.cpp
@@ -215,7 +215,7 @@ bool AceViewSG::DispatchBasicEvent(const std::vector<TouchEvent>& touchEvents)
             continue;
         }
         if (touchEventCallback_) {
-            touchEventCallback_(point, nullptr);
+            touchEventCallback_(point, nullptr, nullptr);
         }
     }
     // if it is last page, let os know to quit app
@@ -234,7 +234,7 @@ bool AceViewSG::DispatchTouchEvent(const std::vector<uint8_t>& data)
             continue;
         }
         if (touchEventCallback_) {
-            touchEventCallback_(point, nullptr);
+            touchEventCallback_(point, nullptr, nullptr);
         }
     }
     // if it is last page, let os know to quit app
